### DISCUSSION TITLE
Preventing or minimising defrost cycles?

### DISCUSSION THREAD

**Discussion by martindell** [Arnold-n/P1P2MQTT#142] [Q&A]
It just got cold here (in the UK) and my Daikin Altherma spent the whole day cycling in and out of defrost cycles.  Thanks to Arnold's magic P1/P2 solution - at least I can see what's happening.  Comparing outside temperature and power draw on the compressor unit - it seems to be trying to preempt freezing up when temp drops below 2 degrees C. (Between 18:00 and 18:30 is when I experimented with blasting blast heat into the house - without much success.)

![screencapture-mm8a9v0rs4r6ouggtwh0e6h6m8866cp5-ui-nabu-casa-history-2024-11-19-20_22_29-edit](https://github.com/user-attachments/assets/19d84dfd-c5e7-4692-b449-1d849fa00255)

Comparing top chart (outdoor temp) with bottom chart (outdoor unit power), this waste of energy started between 10:00 and 10:30 when temp dropped - and carried on all day despite the house not demanding heat at all.

I'll take a look at the heat exchanger fins tomorrow to see if they're clean, but googling around, I came across this article about hacking Daikin's defrost cycles: https://harizanov.com/2012/02/defrost/

It relies on being able to reset the indoor unit (I think) before it starts defrosting.  Finally, I have two questions:

1. Can the P1/P2 box issue a restart command somehow?
2. Has anyone else coped with Daikin's aggressive defrosts some other way?

**Comment by Arnold-n** [comment]
Hi @martindell - nice to hear from you again. Assuming you updated the ATmega firmware to 0.9.28 or later, you can request the Daikin to restart using the command L99 or (with recent firmwares) with HA button `Daikin_Restart_Careful`. This is intended to restart after a field setting change, I am not sure if it is safe to do this often. A restart is disabled by the bridge if the compressor is on even though Daikin's controllers do not seem to care about that. 

One way of minimizing defrosts is by reducing LWT when it is cold.

In your message I am confused that the heat pump is running at all if you indicate that the house is not demanding any heat at all; could it be cycling due to LWT overshoot instead of due to defrosts?

**Reply by martindell** [reply]
Hi Arnold - P1/P2 just works - so I haven't dared to update it!  I'm so grateful for it - especially with these kinds of challenges.  

Yes, it's weird.  No heat demand from the house at all between 10:00 and 18:00 but a lot of compressor cycling.  I'm calling it defrost because I don't know what else it could be?  

Right now, the indoor unit pump is running (even though all thermostats are OFF) and it says it's in space heating mode.  Every now and then it'll fire up the compressor too - again with no heat demand from the house

![Screenshot 2024-11-19 at 21 34 46](https://github.com/user-attachments/assets/372844ed-fb4a-40b0-ac07-5c5b6a56925b)

I preheat the house between 23:30 and 05:30 on cheap electricity and so far, it has stayed warm enough by the time we go to bed. But this compressor cycling between 10:00 and 18:00 is new.  The only thing that's changed is that the temperature has dropped outside (not so much in the house).

![Screenshot 2024-11-19 at 21 37 50](https://github.com/user-attachments/assets/c06b86a8-f9f4-4adc-9bd1-7596405ce1e3)

Have you ever seen similar behaviour?

**Reply by Arnold-n** [reply]
One advantage of updating would be that the bridge now reports when defrosts take place. I doubt you are seeing (only) defrosts here - defrosts have relatively short compressor runs and should be recognizable as LWT dips shortly below RWT, and flow is maximal.

How does your heatpump "know" the house does not have any heat demand? Usually your Daikin thermostat is either running in  room thermostat mode, or the system is controlled by a binary heat demand input signal - which also needs to be configured in the heat pump ([C-07] = 1 for "ext RT control" - unit operation is decided by the external thermostat) otherwise the heat pump always want to keep LWT at its setpoint (which may explain why you see the compressor kick in when outside temp is lower -> LWT setpoint is higher).

**Reply by martindell** [reply]
Yes, I'm using ext RT control to call for heat / run the pump.  I started the update but have done something stupid and now can't access it at all.  Waiting for an ESP-01 programmer to arrive ..

I checked the outside unit and it did have frost on the coils.  I cleaned them with a brush just in case, but am flying blind now ..

**Reply by Arnold-n** [reply]
Oops - for your hardware version (only for V1.0) you will need a modified ESP-01 programmer (shortening of the male pins), which was included in your original shipment (hope you can find it?). If you use an unmodified programmer, please ensure that the male pins of the programmer remain isolated from the ESP casing.

If the bridge cannot connect to WiFi, it should set up an AP for you; if it can connect to your WiFi but cannot connect to your MQTT server any more, you can try to telnet to it and set the MQTT settings again.

I do not know how to prevent defrosts (often they are not really needed), but the bridge can request the unit to defrost, though Daikin's documentation indicates that the unit ultimately decides whether to grant such a defrost request.

**Reply by martindell** [reply]
Thanks Arnold, I can read sensors but can’t control anything from the P1/P2 unit. I guess that means the monitor is working but the bridge is not? I ran out of time today to test much more, but I think one of the Daikin units has some kind of a problem. I’m waiting for a service visit from Daikin

**Reply by Arnold-n** [reply]
As you noticed, "L1" solves this. If the bridge detects errors (for example by disconnecting live) it may switch back to "L0" for safety reasons.

**Comment by martindell** [comment]
Hi @Arnold-n I'm making progress.  Connected OK via telnet, reset MQTT details, issued an L1 command and now I have control again - phew.  Even though telnet works, I can't seem to update to the latest firmware:
```
P1P2/S/040 * [ESP] MQTT_server is 192.168.0.20
P1P2/S/040 * [ESP] MQTT_port is 1883
P1P2/S/040 * [ESP] MQTT_user is MQTT
P1P2/S/040 * [ESP] mqttInputByte4 is 0
P1P2/S/040 * [ESP] hwID 0
P1P2/S/040 * [ESP] noWiFi is 0
P1P2/S/040 * [ESP] Local IP address: 192.168.0.40
P1P2/S/040 * [ESP] Uptime 15130
``` 

```
~/.arduino15/packages/esp8266/tools/python3/3.7.2-post1/python3 -I ~/.arduino15/packages/esp8266/hardware/esp8266/3.0.2/tools/espota.py -i <IPv4> -p 8266 --auth=P1P2MQTT -f <bin-file>
``` 
```
20:35:00 [INFO]: Starting on 0.0.0.0:45196
20:35:00 [INFO]: Upload size: 816912
20:35:00 [INFO]: Sending invitation to: 192.168.0.40
Authenticating...OK
20:35:00 [INFO]: Waiting for device...
20:35:10 [ERROR]: No response from device1P2-bridge-esp8266 to convert the P1/P2 data to MQTT (or telnet) via WiFi and accept commands  over MQTT or telnet
``` 
Am I doing something dumb?  The ESP-01 programmer will arrive tomorrow so I can try that instead

**Reply by Arnold-n** [reply]
That should work - but older firmware versions could suffer from workload under less than ideal WiFi conditions. It may help to reduce the workload ("J0") and/or to restart the bridge first.

If you update with the ESP-01, you will lose all settings so the AP will start again, and (this is for V1.0 bridges only) you will have to tell the bridge it is V1.0 (command "P31 0" followed by "D5).

After the ESP update, you will also need to update the ATmega with avrdude (which requires hwID to be 0, if needed by the "P31 0"/"D5" action, for V1.0 only).

**Reply by martindell** [reply]
OK thanks - no joy, I'll wait for the ESP-01

**Comment by martindell** [comment]
Well, I managed for 'fix' the symptom - but not the underlying problem.  I set the minimum LWT to 20 and now the pump runs / samples but doesn't switch the compressor on - because the water temp is above 20.  

What's weird is that the pump is set to request mode but this sensor: sensor.s1_circulation_pump_onoff is set to 1 permanently and sensor.s0_heating_onoff = 1 even when LWT is OFF

**Comment by Arnold-n** [comment]
I would recommend to set the LWT setpoint at 25C or above. If the LWT setpoint is below 25C, the compressor is never used - I do not think the Daikin manual warns explicitly about this but it is a small note below the operation range graphs in the installer reference guide.

**Comment by martindell** [comment]
Interesting (for me anyway) that this is an outside temp-related 'feature'
![Screenshot 2024-11-21 at 09 19 19](https://github.com/user-attachments/assets/292548e1-3315-4c8a-8ca9-287dd4f16d58)
Before 15th Nov, the pump ran only during LWT / DHW operation. On the 15th when temp dipped below 2 degrees, this constant pump-sampling started to happen - but it was too quick for me to notice. I only noticed on the 19th when temp dropped consistently below 2 degrees.  I have no idea what this operation is aiming to achieve - it doesn't seem related to any Daikin setting I can see?

**Reply by Arnold-n** [reply]
Could it be related to pipe anti-frost protection? Usually this is off ([4-04]=2) on non-monoblock systems.

**Comment by martindell** [comment]
Yeah, my 4-04 = 2 :/ Nice idea though, I wonder what the values of 0 and 1 would do?  No info in my installation manual
- found this online: 0- continuous,  1- intermittent,   2-no protection

**Comment by martindell** [comment]
Spoke to a Daikin tech today and he confirmed that it is indeed an internal pipe-freeze thing.  And it occurred to me that maybe my indoor unit doesn’t know what the temp is indoors - I ditched the Daikin thermostat years ago.  Will check my settings when I get home.

Update to the latest MQTT worked via the ESP-01. So many new sensors!  I played a little bit with controlling DHW and LWT but the on/off commands didn’t seem to work every time.  I restarted the device a few times but that didn’t seem to help.  Are there any settings / logs I should check?

**Comment by Arnold-n** [comment]
It is helpful to monitor MQTT topic P1P2/S/# or observe messages from a telnet session. Sounds like you are running out of write budget.

**Comment by martindell** [comment]
Ah, that makes sense - will do, thanks.

**Comment by martindell** [comment]
Mystery solved.  I had a rogue setting for internal thermostat. I set 0c 7 to 1 and the pump cycling stopped. Thanks for your help @Arnold-n

**Comment by savage7** [comment]
@martindell I have a similar problem and followed this discussion. What do you mean with `0c 7 to 1 `?

**Comment by martindell** [comment]
Hi, in the installer settings on my unit, there's a set of cryptic settings
that Arnold exposes as field settings.  On my one, page 0c, position 7 was
set to 2 - and I changed it to 1.

My hypothesis is that the unit had entered a pipe freeze prevention mode
because of the lower outside temperature.

Originally, my system had a Daikin room thermostat, but I replaced it last
year with smart thermostats and TRV's.

My guess is that the heat pump was running a frost prevention mode because
it no longer had a measurement for internal temperature.

Changing this setting was my best guess - and it seemed to work.

**Comment by Arnold-n** [comment]
Makes sense, to add: field setting 0c-7 (Daikin denotes this as [C-07], the P1P2MQTT setting starts with C_07 to avoid a minus signal in the name) relates to system control by either 0:LWT (always on), 1:external room thermostat (relays), or 2:room thermostat. Seems indeed that some protection is triggered by the combination of room thermostat mode and missing room temperature sensor.
